aws/resource_aws_instance.go

```diff
@@ -1066,6 +1117,35 @@ func InstanceStateRefreshFunc(conn *ec2.EC2, instanceID, failState string) resou
 	}
 }
 
+// VolumeStateRefreshFunc returns a resource.StateRefreshFunc that is used to watch
+// an EC2 root device volume.
+func VolumeStateRefreshFunc(conn *ec2.EC2, volumeID, failState string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		resp, err := conn.DescribeVolumesModifications(&ec2.DescribeVolumesModificationsInput{
+			VolumeIds: []*string{aws.String(volumeID)},
+		})
+		if err != nil {
+			if ec2err, ok := err.(awserr.Error); ok && ec2err.Code() == "InvalidVolumeID.NotFound" {
+				resp = nil
```

Comment:
Why can't we just `return nil, "", nil` here? That would also simplify the code below as we can then omit the else block and conditional below. 😉